src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -80,11 +102,29 @@ public RoutingResponse route() {
 
         debugTimingAggregator.finishedRouting();
 
+        Instant filterOnLatestDepartureTime = null;
+
+        // Filter itineraries away that depart after the latest-departure-time for depart after
+        // search. These itineraries are a result of time-shifting the access leg and is needed for
+        // the raptor to prune the results. These itineraries are often not ideal, but if they
+        // pareto optimal for the "next" window, they will appear when a "next" search is performed.
+        if(!request.arriveBy
+            && raptorSearchParamsUsed != null
+            && raptorSearchParamsUsed.isSearchWindowSet()
+            && raptorSearchParamsUsed.isEarliestDepartureTimeSet()
+        ) {
+            int ldt = raptorSearchParamsUsed.earliestDepartureTime()
+                    + raptorSearchParamsUsed.searchWindowInSeconds();
+            filterOnLatestDepartureTime = searchStartTime.plusSeconds(ldt).toInstant();
+        }
+
+
```

Comment:
I do not think this have the same abstraction level as the rest of the method, so maybe extract this into a method called `filterOnLatestDepartureTime()` and inline the variable.